cpu/cc26x0/cpu.c

```diff
@@ -17,13 +17,47 @@
  * @}
  */
 
-#include "cpu.h"
+ #include "cpu.h"
+ #include "periph_conf.h"
 
-/**
- * @brief Initialize the CPU, set IRQ priorities
- */
-void cpu_init(void)
-{
-    /* initialize the Cortex-M core */
-    cortexm_init();
-}
+ static void cpu_clock_init(void);
+
+ /**
+  * @brief Initialize the CPU, set IRQ priorities
+  */
+ void cpu_init(void)
+ {
+     /* initialize the Cortex-M core */
+     cortexm_init();
+
+     /*initialize the system clock*/
+     cpu_clock_init();
+ }
+
+ /**
+   *@brief Configure the MCU system clock
+   */
+ static void cpu_clock_init(void)
+ {
+   AON_WUC->AUXCTL |= AUXCTL_AUX_FORCE_ON; /* power on AUX_PD */
+   while(!(AON_WUC->PWRSTAT & PWRSTAT_AUX_PD_ON)); /* wait for AUX_PD to be powered on */
+   AUX_WUC->MODCLKEN0 |= MODCLKEN0_AUX_DDI0_OSC_EN; /*turn on oscillator interface clock */
+
+   /* configure HF clock */
+   #if HF_CLOCK_HF_ROSC
+     DDI_0_OSC->CTL0 |= 0x0; /* set 48MHz RCOSC */
+   #elif HF_CLOCK_HF_XOSC
+     DDI_0_OSC->CTL0 |= DDI_0_OSC_CTL0_SCLK_HF_SRC_SEL; /* set 48MHz XOSC (24MHz doubled internally) */
+   #endif
+
+   /* configure LF clock */
+   #if LF_CLOCK_HF_RCOSC
+     DDI_0_OSC->CTL0 |= DDI_0_OSC_CTL0_SCLK_LF_SRC_SEL_HF_RCOSC; /* set 31.25kHz derived from 48MHz RCOSC */
+   #elif LF_CLOCK_HF_XOSC
+     DDI_0_OSC->CTL0 |= DDI_0_OSC_CTL0_SCLK_LF_SRC_SEL_HF_XOSC; /* set 31.25kHz derived from 24MHz XOSC */
+   #elif LF_CLOCK_LF_RCOSC
+     DDI_0_OSC->CTL0 |= DDI_0_OSC_CTL0_SCLK_LF_SRC_SEL_LF_RCOSC; /* set 32kHz derived from 32kHz RCOSC */
+   #elif LF_CLOCK_LF_XOSC
+     DDI_0_OSC->CTL0 |= DDI_0_OSC_CTL0_SCLK_LF_SRC_SEL_LF_XOSC; /* set 32.768kHz derived from 32.768kHz XOSC */
+   #endif
+ }
```

Comment:
same here.

these are many lines of code :-/
in the end, the whole should probably look more like this:
```
#ifndef HF_CLOCK_SOURCE
#define HF_CLOCK_SOURCE DDI_0_OSC_CTL0_SCLK_HF_SRC_SEL_RCOSC
#endif
#ifndef LF_CLOCK_SOURCE
#define LF_CLOCK_SOURCE DDI_0_OSC_CTL0_ACLK_REF_SRC_SEL_RCOSC_HF
#endif
...
static void cpu_clock_init(void)
...
DDI_0_OSC->CTL0 |= HF_CLOCK_SOURCE | LF_CLOCK_SOURCE;
```